entry-asar/has-asar.js

```diff
@@ -1,7 +1,24 @@
 if (process.arch === 'arm64') {
-  process._archPath = require.resolve('../app-arm64.asar');
+  setPaths('arm64');
 } else {
-  process._archPath = require.resolve('../app-x64.asar');
+  setPaths('x64');
+}
+
+function setPaths(platform) {
+  // This should return the full path, ending in something like
+  // Notion.app/Contents/Resources/app.asar
+  const appPath = app.getAppPath()
+  const asarFile = `app-${platform}.asar`
+
+  // Maybe we'll handle this in Electron one day
+  if (appPath.includes('app.asar')) {
+    const platformAppPath = appPath.replace(/\.app\.asar$/, asarFile)
+
+    // This is an undocumented API. It exists.
+    app.setAppPath(platformAppPath)
```

Comment:
Maybe add a comment in Electron code that it's used by universal? In case someone decides to do some early spring cleaning